### PR TITLE
Add "Feature Interactions" section to SIP template

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**

### CODE REVIEW

**Comment by anatoliykmetyuk** [comment]
@bjornregnell what do you think of this new section?

File: sip-template.md [hunk 1]
```diff
@@ -62,6 +62,14 @@ If it doesn't do so "by construction", this section should present the ideas of
 
 This section should also argue to what extent backward source compatibility is preserved. In particular, it should show that it doesn't alter the semantics of existing valid programs.
 
+### Feature Interactions
+
+A discussion of how the proposal interacts with other language features. Think about the following questions:
+
+- When envisioning the application of your proposal, what features come to mind as the most likely to be interacting with it?
```
Comment (SethTisue) [inline sip-template.md:69]:
```suggestion
- When envisioning the application of your proposal, what features come to mind as most likely to interact with it?
```

**Comment by anatoliykmetyuk** [comment]
> Looks good! Perhaps say something about trying to come up with proposals on how to solve any identified problematic interaction: a) how should it work, limitations/checks/restrictions on syntax/semantics, b) compiler should not allow this combination and give error message

Makes sense, I've added a clarifying sentence to mention those possible ways to solve such interactions.

**Review by bjornregnell — APPROVED** [review]
Nice! LGTM

**Pull request merged**